app/src/main/java/com/aptoide/uploader/apps/InstalledApp.java

```diff
@@ -1,39 +1,217 @@
+    this.isSystemApp = systemApp;
+  }
+
+  public String getApkPath() {
+    return apkPath;
+  }
+
+  public void setApkPath(String apkPath) {
     this.apkPath = apkPath;
+  }
+
+  public String getIconPath() {
+    return iconPath;
+  }
+
+  public void setIconPath(String icon) {
+    this.iconPath = icon;
+  }
+
+  public long getInstalledDate() {
+    return installedDate;
+  }
+
+  public void setInstalledDate(long installedDate) {
     this.installedDate = installedDate;
-    this.versionCode = versionCode;
-    this.isUploaded = isUploaded;
+  }
+
+  public Obb getObbMain() {
+    return obbMain;
+  }
+
+  public void setObbMain(Obb obbMain) {
     this.obbMain = obbMain;
+  }
+
+  public Obb getObbPatch() {
+    return obbPatch;
+  }
+
+  public void setObbPatch(Obb obbPatch) {
     this.obbPatch = obbPatch;
   }
 
+  public boolean isUploaded() {
+    return isUploaded;
+  }
+
+  public void setIsUploaded(boolean uploaded) {
+    isUploaded = uploaded;
+  }
+
+  public int getStatus() {
+    return status;
+  }
+
+  public void setStatus(int status) {
+    this.status = status;
+  }
+
+  public ApplicationInfo getAppInfo() {
+    return appInfo;
+  }
+
+  public void setAppInfo(ApplicationInfo appInfo) {
+    this.appInfo = appInfo;
+  }
+
+  private Obb getMainObb(String packageName) {
+    String sdcard = Environment.getExternalStorageDirectory()
+        .getAbsolutePath();
+    File obbDir = new File(sdcard + "/Android/obb/" + packageName + "/");
+    if (obbDir.isDirectory()) {
+      File[] files = obbDir.listFiles();
+      if (files != null) {
+        Arrays.sort(files, (f1, f2) -> Long.compare(f2.lastModified(), f1.lastModified()));
+        for (File file : files) {
+          if (file.getName()
+              .contains("main") && !file.getName()
+              .contains("--downloading")) {
+            String obbMainPath = file.getAbsolutePath();
+            return new Obb(obbMainPath.substring(obbMainPath.lastIndexOf("/") + 1), null,
+                obbMainPath);
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  private Obb getPatchObb(String packageName) {
+    String sdcard = Environment.getExternalStorageDirectory()
+        .getAbsolutePath();
+    File obbDir = new File(sdcard + "/Android/obb/" + packageName + "/");
+    if (obbDir.isDirectory()) {
+      File[] files = obbDir.listFiles();
+      if (files != null) {
+        Arrays.sort(files, (f1, f2) -> Long.compare(f2.lastModified(), f1.lastModified()));
+        for (File file : files) {
+          String arr[] = file.getName()
+              .split("\\.", 2);
+          if (arr[0].equals("patch") && !file.getName()
+              .contains("--downloading")) {
+            String obbPatchPath = file.getAbsolutePath();
+            return new Obb(obbPatchPath.substring(obbPatchPath.lastIndexOf("/") + 1), null,
+                obbPatchPath);
+          }
+        }
+      }
+    }
+    return null;
+  }
+
```

Comment:
Is this needed?

Reply:
Most of the getters and setters already existed, but now are being used in the constructor to set the variables. I can remove the setters and use normal variable attribution in the constructor but though it was cleaner this way. 
Should i change it back to the way it was?

Reply:
This part is the same as the discussion above.